### ISSUE TITLE
VSCode ate my code! (data loss on crash)

### ISSUE THREAD

**Issue by mpartel** [Microsoft/vscode#50368]
- VSCode Version: 1.23.1
- OS Version: Ubuntu 16.04

VSCode hung. I killed it with `xkill`. Upon reopening VSCode, the file I was editing was **completely empty** (on disk too, not just the editor).

I was working on C# code. I have autosave enabled (`"files.autoSave": "afterDelay",`). I've never seen a hang like this before despite many months of active use on the same machine and OS.

I went looking into `dmesg` for any disk/filesystem weirdness, but all I saw was `traps: code[22755] general protection ip:7fd8f014754d sp:7ffdfd91d570 error:0 in libnode.so[7fd8ef843000+137e000]` i.e. VSCode segfaulted.

Bottom line: it would be good if VSCode didn't overwrite files on save, but rather write into a new file and move the result over the original. E.g. KWrite works like this.
